resources/utils/displayValue.ts

```diff
@@ -0,0 +1,132 @@
+import BigNumber from 'bignumber.js'
+import { isHexString } from 'ethers/lib/utils'
+
+const displayUnitMapping = {
+  million: {
+    lowerBound: BigNumber(1000000),
+    upperBound: BigNumber(1000000000),
+    unitDisplay: 'M'
+  },
+  billion: {
+    lowerBound: BigNumber(1000000000),
+    upperBound: BigNumber(1000000000000),
+    unitDisplay: 'B'
+  },
+  trillion: {
+    lowerBound: BigNumber(1000000000000),
+    upperBound: BigNumber(1000000000000000),
+    unitDisplay: 'T'
+  },
+  quadrillion: {
+    lowerBound: BigNumber(1000000000000000),
+    upperBound: BigNumber(Infinity),
+    unitDisplay: 'Q'
+  }
+}
+const maxDisplayValue = BigNumber(999999999999999999999)
+
+function getDisplay (bn: BigNumber, type: string, decimals: number, displayFullValue?: boolean) {
+  const value = bn.decimalPlaces(decimals, BigNumber.ROUND_FLOOR)
+  if (value.isZero()) {
+    return {
+      approximationSymbol: '<',
+      displayValue: BigNumber(`1e-${decimals}`).toFormat()
+    }
+  }
+
+  if (!displayFullValue) {
+    // shorthand display of large numbers
+    for (const [unitName, { lowerBound, upperBound, unitDisplay }] of Object.entries(displayUnitMapping)) {
+      if (value.isGreaterThanOrEqualTo(lowerBound) && value.isLessThan(upperBound)) {
+        const displayMax = value.isGreaterThan(maxDisplayValue)
+        // maximum display value is hard coded because maxDisplayValue is above the bignumber 15sd limit
+        return {
+          approximationSymbol: displayMax ? '>' : '',
+          displayValue: displayMax ? '999,999' : value.shiftedBy(-(lowerBound.sd(true) - 1)).decimalPlaces(2, BigNumber.ROUND_FLOOR).toFormat(),
+          displayUnit: {
+            fullName: unitName,
+            shortName: unitDisplay
+          }
+        }
+      }
+    }
+  }
+
+  // display small numbers or full values
+  return {
+    displayValue: value.toFormat(type === 'fiat' ? decimals : undefined)
+  }
+}
+
+type DisplayValueDataParams = { 
+  currencyRate?: Rate
+  displayFullValue?: boolean 
+  decimals: number
+  isTestnet: boolean
+}
+
+type SourceValue = string | number | BigNumber
+
+export function displayValueData (sourceValue: SourceValue, params: DisplayValueDataParams) {
```

Comment:
what do `displayValueData` and `getDisplay` do? maybe a clearer name or some comments can help here